pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -84,9 +129,48 @@ func (r *CloudEntitySelectorReconciler) SetupWithManager(mgr ctrl.Manager) error
 		}); err != nil {
 		return err
 	}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cloudv1alpha1.CloudEntitySelector{}).
-		Complete(r)
+
+	if err := ctrl.NewControllerManagedBy(mgr).For(&cloudv1alpha1.CloudEntitySelector{}).Complete(r); err != nil {
+		return err
+	}
+	return mgr.Add(r)
+}
+
+func (r *CloudEntitySelectorReconciler) Start(context.Context) error {
+	if r.initialized {
+		r.Log.Info(controllerDuplicateInit)
+		return nil
+	}
+	// Is cache synced?
+	if !(*r.Mgr).GetCache().WaitForCacheSync(context.TODO()) {
```

Comment:
what is the timeout?

Reply:
Could not test it, but in scale environments shared informer cache was synced. We did not observe any noticeable delays, so we should be fine.